src/ucp/core/ucp_ep.c

```diff
@@ -885,7 +885,7 @@ ucp_ep_create_api_to_worker_addr(ucp_worker_h worker,
         !(flags & UCP_EP_PARAMS_FLAGS_NO_LOOPBACK)) {
         ucp_ep_update_remote_id(ep, ucp_ep_local_id(ep));
         ucp_ep_flush_state_reset(ep);
-    } else if (conn_sn != UCP_EP_MATCH_CONN_SN_MAX) {
+    } else if (flags & UCP_EP_PARAMS_FLAGS_NO_PEER_MATCH) {
```

Comment:
should insert if not no-peer-match (or invert if/else)

Reply:
fixed